src/callbacks/ltfb.cpp

```diff
@@ -456,7 +456,9 @@ void ltfb::on_batch_begin(model *m) {
   if (comm.am_world_master()) {
     std::cout << message_prefix + "evaluating local model...\n";
   }
-  const auto local_score = evaluate(*m, m_metric_name);
+  auto local_score = evaluate(*m, m_metric_name);
+  if (local_score == 0) { local_score = std::nan(""); } /// @todo REMOVE!
+  if (std::isinf(local_score)) { local_score = std::nan(""); } /// @todo REMOVE!
```

Comment:
It makes sense to me that inf scores are invalid, but zeros could be perfectly sensible, e.g. if we evaluate with categorical accuracy.
```suggestion
  auto local_score = evaluate(*m, m_metric_name);
  if (std::isinf(local_score)) { local_score = std::nan(""); }
```